appsec/src/extension/user_tracking.c

```diff
@@ -61,6 +67,78 @@ static PHP_FUNCTION(set_user_wrapper)
     _ddtrace_set_user(INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+static void _emit_user_event(void)
+{
+    dd_tags_set_user_event_triggered();
+    dd_trace_emit_asm_event();
+    dd_tags_set_sampling_priority();
+}
+
+static PHP_FUNCTION(v2_track_user_login_success_wrapper)
+{
+    _ddtrace_v2_track_user_login_success(INTERNAL_FUNCTION_PARAM_PASSTHRU);
+    if (!DDAPPSEC_G(active) && UNEXPECTED(!get_global_DD_APPSEC_TESTING())) {
+        return;
+    }
+    _emit_user_event();
+    dd_telemetry_add_sdk_event(LSTRARG("login_success"));
+
+    zend_string *login;
+    zval *user = NULL;
+    zend_array *metadata = NULL;
+    zend_string *user_id = NULL;
+    if (zend_parse_parameters(
+            ZEND_NUM_ARGS(), "S|zh", &login, &user, &metadata) == FAILURE) {
+        return;
+    }
+    if (_ddtrace_v2_track_user_login_success == NULL) {
```

Comment:
I'm a bit confused, why do you check this here, when you already called the function before? Same applies to the login failure function